view/src/components/foodbanks.js

```diff
@@ -230,8 +256,425 @@ class Foodbank extends Component {
+                  label="Foodbank Names"
+                  placeholder="Type or Select a Food Bank Name"
+                  variant="outlined"
+                  name="nameQuery"
+                  onChange={(e) => this.handleStringSearch("nameQuery", e)}
+                  InputProps={{
+                    ...params.InputProps,
+                    startAdornment: (
+                      <>
+                        <InputAdornment position="start">
+                          <SearchIcon />
+                        </InputAdornment>
+                        {params.InputProps.startAdornment}
+                      </>
+                    ),
+                  }}
+                />
+              )}
+            />
+          </AccordionSummary>
+          <AccordionDetails>{this.extraFiltersAccordion()}</AccordionDetails>
+          <Grid container alignItem="left">
+            <Grid item xs={12}>
+              <Box paddingLeft={2} paddingBottom={2}>
+                <Button
+                  className={classes.clearSearchButton}
+                  onClick={this.resetCards}
+                  variant="contained"
+                  color="primary"
+                  size="medium"
+                  startIcon={<ClearIcon />}
+                >
+                  Clear
+                </Button>
+              </Box>
+            </Grid>
+          </Grid>
+        </Accordion>
+      </div>
+    );
+  };
+
+  /**
+   * Returns tag filtering menu for searching cards.
+   * Re-using the autocomplete for the add new/edit item form
+   * causes issues with setting a conditional value for
+   * Autocomplete value={}, so separating the two is smoother.
+   */
+  tagsAutocomplete = () => {
+    const { classes } = this.props;
+    const { allTags, tagsQuery } = this.state;
+
+    return (
+      <Autocomplete
+        className={classes.searchBars}
+        multiple
+        id="tags-search"
+        onChange={this.handleTagFilter}
+        options={allTags}
+        value={tagsQuery}
+        fullWidth
+        renderTags={(value, getTagProps) =>
+          value.map((option, index) => (
+            <Chip label={option} {...getTagProps({ index })} />
+          ))
+        }
+        renderInput={(params) => (
+          <TextField
+            {...params}
+            variant="outlined"
+            label="Food Bank Tags"
+            placeholder="Type or Select a Food Bank Tag"
+            InputProps={{
+              ...params.InputProps,
+              startAdornment: (
+                <>
+                  <InputAdornment position="start">
+                    <SearchIcon />
+                  </InputAdornment>
+                  {params.InputProps.startAdornment}
+                </>
+              ),
+            }}
+          />
+        )}
+      />
+    );
+  };
+
+  /** Updates filteredData with a new array; used with filters.js queries */
+  updateCards = (newValues) => {
+    if (newValues.length === 0) {
```

Comment:
can newValues ever be undefined? if so, trying to access .length won't go well

Reply:
hmm so it should always be at least an empty array, since it gets returned from two filters.js functions that use a default [] value i updated it to if (newValues && newValues.length === 0), to check for undefined situations to be safer. i confirmed that the [] is truthy in react, and added a note to update the checks if we use updateCards for more things in the future